### PR TITLE
BitBucket Server Provider

### PR DESCRIPTION
**Pull request by cryptic-ai**
Created a new provider (per last PR comment). I did take a look to see if I could get the origin from the headers, and while I can get the origin in which Skyhook is running, I'm not able to get the Bitbucket Server URL. You have to set the Bitbucket Server url using the `SERVER` environment variable. I'll keep updating this PR as methods are added.

- [x] Mirrored repository synchronized
- [x] Pull request approved
- [x] Pull request comment added
- [x] Pull request comment deleted
- [x] Pull request comment edited
- [x] Pull request declined
- [x] Pull request deleted
- [x] Pull request marked needs work
- [x] Pull request merged
- [x] Pull request modified
- [x] Pull request opened
- [x] Pull request reviewer updated
- [x] Pull request unapproved
- [x] Repository commit comment added
- [x] Repository commit comment deleted
- [x] Repository commit comment edited 
- [x] Repository forked
- [x] Repository modified
- [x] Repository refs updated
- [x] Diagnostic ping

### CODE REVIEW

File: src/provider/BitBucketServer.ts [hunk 1]
```diff
@@ -51,9 +51,54 @@ class BitBucketServer extends BaseProvider {
...
+        toFromRefField.name = this.body.pullRequest.title
+        toFromRefField.value = `**Destination branch:** ${this.body.pullRequest.toRef.displayId} \n **State:** ${this.body.pullRequest.state}`
+        fieldArray.push(toFromRefField)
+
+        this.body.pullRequest.reviewers.forEach((reviewer) => {
+            const reviewerField = new EmbedField()
+            reviewerField.name = 'Reviewer'
+            reviewerField.value = reviewer.user.displayName
+            fieldArray.push(reviewerField)
+        })
```
Comment (cryptic-ai) [inline src/provider/BitBucketServer.ts:98]: Will probably need to do a check and only pull out a max of 18 to make sure we don't breach the 20 field limit. Open to suggestions on how to better do that.

Comment (cryptic-ai) [reply src/provider/BitBucketServer.ts:98]: Could potentially do something like 
```ts
       let reviewerNames = ''
       const reviewerField = new EmbedField()
       reviewerField.name = 'Reviewers'
       reviewerField.value = reviewer.user.displayName
       fieldArray.push(reviewerField)
        this.body.pullRequest.reviewers.forEach((reviewer) => {
            reviewerNames = reviewerNames + '${reviewer.user.displayName} \n'
        })
```

**Comment by cryptic-ai** [comment]
All webhooks have been added and tested from my local instance. If I need to write more tests other than the one that's in there, please let me know! I look forward to your review.

**Comment by cryptic-ai** [comment]
This should fix #141 if accepted

**Comment by dscalzi** [comment]
You're sure the origin header is not provided in the req object here? https://github.com/Commit451/skyhook/blob/master/src/index.ts#L104

Obviously using the environment variable will work if you self host this, but it would not work on the public hosted version.

**Comment by cryptic-ai** [comment]
Updated to dynamically pull the baseLink so that it can work from the hosted version, updated the tests.

**Pull request merged**